packages/core-data/src/hooks/use-entity-record.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * WordPress dependencies
+ */
+import { useQuerySelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '../';
+import { Status } from './constants';
+
+interface EntityRecordResolution {
+	/** The requested entity record */
+	record: Object;
```

Comment:
`Object` is not a type that we'd like to use in this context, it behaves similarly to `any` that it just silent the errors without any actual gains.

Ideally, we should already have all the types of core entities available somewhere so that we can do this automatically. Given that it'd be a huge amount of work, we could instead try to use [generics](https://www.typescriptlang.org/docs/handbook/2/generics.html) here to let the users specify the correct types.

```ts
const { record } = useEntityRecord<PageRecord>( 'postType', 'page', id );
// record: PageRecord
```

It's a bit more complicated to get right though, I'm willing to help if you encounter any problems!

Reply:
Also, if the data is still resolving, I think the type of this value will be `null`?

Reply:
Hey @kevin940726, you will like this PR: [TypeScript definitions for core-data entity records](https://github.com/WordPress/gutenberg/pull/38666)